README.md

```diff
@@ -1,6 +1,199 @@
+  - `scripts/ci/.gitlab-ci.platform.yaml`
+  - `venv`
+
+   > @TODO: Include new files in the build directory but not in the artifact and note deleted files from the artifact.
+
+---
+
+### Propagating Versions
+
+Once you have the list of different files, filter out platform resources (to be covered in more detail). With the help of the inventory helper (which checks resources and variables), resources are divided into namespaces. These namespaces are split into **domain** and **packages**. While the domain is always static, packages come from `plasma-compose`.
+
+#### Syncing Resource Versions Across Namespaces:
+Next, we filter out resources that exist in multiple namespaces. `compose` handles this via merge strategies, but `sync` compares the build version with the namespace versions to select the correct resource. Once sorted, we iterate over the list of resources that differ between the build and the artifact.
+
+> @TODO: (Temporary) Should we include additional technical details here?
+
+### Iterating Through Resources:
+During this phase, the following rules apply:
+- If a resource is deleted from the build, it’s skipped. The developer should handle dependencies manually and bump related resources.
+- If a resource is new, add it to the propagation list.
+- If a resource differs between the build and the artifact, compare versions. If the base version (e.g., base, base-propagation_suffix) differs, add it to the propagation list. Otherwise, copy the resource version from the artifact.
+
+The same process is applied to variables by comparing `group_var` and `vault.yaml` files. Deleted, changed, or added variables will also be propagated.
+
+#### Propagation Map and Timeline:
+To propagate, we first search for the commit where the resource version was set, and a special list of objects called `timeline` (temporary name) is created. Each timeline object contains the commit date, a list of variables to propagate, or a list of resources. The timeline is sorted by date to maintain the correct version sequence as if propagation was applied after each change.
+
+After iterating through the timeline and applying dependent resource versions, the latest state is achieved. At this point, we update the real resource in the build directory.
+
+- If the resource version matches the propagation map, it’s skipped.
+- Otherwise, the version is set as `resource_version-propagated_version`.
+
+---
+
+### Common Propagation Cases (WIP — Convert to Tests?)
+
+#### Case 0:
+As a developer:
+- Clone an existing git repo.
+- Run `plasmactl bump` to update the resource version.
+- Run `plasmactl compose --conflicts-verbosity --skip-not-versioned` to prepare the build.
+- Run `plasmactl bump --sync` to propagate versions of updated resources to dependent resources.
+
+**Expected Outcome:**  
+No bumping or propagation occurs. However, the command should copy propagated versions from the artifact if the base version doesn’t match between the build and artifact.
```

Comment:
No bumping occurs because no resource was updated. No propagation occurs because there is no difference between current build and comparison artifact. However, all propagated versions from the artifact sould be copied to current build if the base version doesn’t match between the build and artifact.